metrics/cli/test/deploy/deployer_test.dart

```diff
@@ -430,6 +436,276 @@ void main() {
+        when(firebaseService.enableAnalytics(any)).thenThrow(stateError);
+
+        await expectLater(deployer.deploy(), throwsStateError);
+
+        verify(directory.deleteSync(recursive: true)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() enables the Firebase Analytics before deploying to the hosting",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        whenDirectoryExist().thenReturn(true);
+
+        await deployer.deploy();
+
+        verifyInOrder([
+          firebaseService.enableAnalytics(any),
+          firebaseService.deployHosting(any, any, any),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() initializes the Firestore data",
+      () async {
+        whenCreateGCloudProject().thenAnswer((_) => Future.value(projectId));
+        whenGetDirectory().thenReturn(directory);
+        whenDirectoryExist().thenReturn(true);
+
+        await deployer.deploy();
+
+        verify(firebaseService.initializeFirestoreData(projectId)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() deletes the temporary directory if the Firebase service throws during the Firebase Analytics enabling",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        whenDirectoryExist().thenReturn(true);
+        when(firebaseService.initializeFirestoreData(any))
+            .thenThrow(stateError);
+
+        await expectLater(deployer.deploy(), throwsStateError);
+
+        verify(directory.deleteSync(recursive: true)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() initializes the Firestore data before deploying to the hosting",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        whenDirectoryExist().thenReturn(true);
+
+        await deployer.deploy();
+
+        verifyInOrder([
+          firebaseService.initializeFirestoreData(any),
+          firebaseService.deployHosting(any, any, any),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() configures the Firebase auth providers",
+      () async {
+        whenCreateGCloudProject().thenAnswer((_) => Future.value(projectId));
+        whenGetDirectory().thenReturn(directory);
+        whenDirectoryExist().thenReturn(true);
+
+        await deployer.deploy();
+
+        verify(firebaseService.configureAuthProviders(projectId)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() deletes the temporary directory if the Firebase service throws during the Firebase auth providers configuration",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        whenDirectoryExist().thenReturn(true);
+        when(firebaseService.configureAuthProviders(any)).thenThrow(stateError);
+
+        await expectLater(deployer.deploy(), throwsStateError);
+
+        verify(directory.deleteSync(recursive: true)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() configures the Firebase auth providers before updating the Web config file",
```

Comment:
Would it be better to test that the `deploy` method updates the Metrics config with the google client id returned from the `configureAuthProviders` method?